core/dbt/context/manifest.py

```diff
@@ -46,9 +55,15 @@ def _get_namespace_builder(self) -> MacroNamespaceBuilder:
             None,
         )
 
+    # This will only work at top-level.
     def to_dict(self):
         dct = super().to_dict()
-        dct.update(self.namespace)
+        # This moves all of the macros in the 'namespace' into top level
+        # keys in the manifest dictionary
+        if isinstance(self.namespace, TestMacroNamespace):
```

Comment:
👀 surprised to see this - going to keep scrolling and might revisit this diff

Reply:
Do you still have questions about that? The part about moving all the macros in the 'namespace' into top level keys is the way it worked (which was a part of the performance issues). The difference here is that using the TestMacroNamespace we only copy the macro we know about and let the the others (if any) be resolved by the dispatch code.

Reply:
yep - I think this checks out now. Thanks for revisiting - all good on my end :)